### PR TITLE
don't need len around shape

### LINKED ISSUE #34274: BUG: small error in pandas/io/pytables.py
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
# Your code here


df.to_hdf('file.h5','name', format='table')

~/miniconda3/lib/python3.6/site-packages/pandas/io/pytables.py in _maybe_convert_for_string_atom(name, block, existing_col, min_itemsize, nan_rep, encoding, errors)
   4798         # we cannot serialize this data, so report an exception on a column
   4799         # by column basis
-> 4800         for i in range(len(block.shape[0])):
   4801 
   4802             col = block.iget(i)

TypeError: object of type 'int' has no len()

```

#### Problem description

I don't know why exactly pandas cannot serialize my data frame with a multi-index.
But the above error prevents the real error message to be printed.

#### Expected Output

Error message. 

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 16.7.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 46.1.3.post20200325
Cython           : 0.29.18
pytest           : 5.4.1
hypothesis       : 5.15.0
sphinx           : 3.0.3
blosc            : None
feather          : None
xlsxwriter       : 1.2.8
lxml.etree       : 3.8.0
html5lib         : 1.0.1
pymysql          : 0.9.3
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.14.0
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fastparquet      : None
gcsfs            : None
lxml.etree       : 3.8.0
matplotlib       : 3.2.1
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 5.4.1
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.17
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.2.8
numba            : 0.48.0

</details>

### PR DESCRIPTION
**Pull request by SilasK**
#Fix #34274

- [x] closes #34274
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Thanks @SilasK - could you add a test to show how this fixes the issue?

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -4789,7 +4789,7 @@ def _maybe_convert_for_string_atom(
 
         # we cannot serialize this data, so report an exception on a column
         # by column basis
-        for i in range(len(block.shape[0])):
+        for i in range(block.shape[0]):
```
Comment (jreback) [inline pandas/io/pytables.py:4792]: can you show what the `block.shape` is in the test suite now (pick a couple of example tests).

**Comment by jreback** [comment]
@SilasK this needs a test where this breaks / why you are changing

**Comment by TomAugspurger** [comment]
@SilasK can you add a test? https://github.com/pandas-dev/pandas/issues/34645 has an example.

**Comment by TomAugspurger** [comment]
@SilasK do you have time to add a test?

**Comment by SilasK** [comment]
Sorry, not for the moment. And It would be a test that test that the error message is produced correctly, how do I do that?

**Comment by MarcoGorelli** [comment]
like this: https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#testing-warnings

**Comment by simonjayhawkins** [comment]
@SilasK Thanks for the PR. closing as stale. LMK if you want to continue and I'll reopen.

**Pull request closed without merging**